### PR TITLE
feat: [CDS-81136] Move K8s BG swap service to be AsyncExecutable

### PR DESCRIPTION
**Pull request by alebuh**
### Test cases
- [x] 1. When FF is disabled use sync (V1) step implementation
- [x] 2. When FF is enabled use async (V2) step implementation
 - Execution was successful
 - The icon should be mapped to the new step type on the UI
 - Logs are not present during the execution, only at the end of it
 - Logs for delegate selector is present
 - Step outputs are populated correctly -> N/A
- [x] 3. Step execution failure - Exceptions have been handled by the Error framework
- [x] 4. Step execution abortion - If the Step is aborted, the task should be aborted on the delegate side as well
 - Logs are missing if the step is aborted
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54242)

### CODE REVIEW

**Comment by alebuh** [comment]
trigger uts

**Comment by alebuh** [comment]
trigger uts

File: 125-cd-nextgen/src/main/java/io/harness/cdng/creator/plan/steps/K8sBGSwapServicesStepPlanCreator.java [hunk 1]
```diff
@@ -53,4 +61,24 @@ protected StepParameters getStepParameters(PlanCreationContext ctx, K8sBGSwapSer
 
     return stepParameters;
   }
+
+  @Override
+  public StepType getStepSpecType(PlanCreationContext ctx, K8sBGSwapServicesStepNode stepElement) {
+    if (featureFlagService.isEnabled(
+            ctx.getMetadata().getAccountIdentifier(), FeatureName.CDS_K8S_ASYNC_STEP_STRATEGY)) {
+      return K8sBGSwapServicesStepV2.STEP_TYPE;
+    } else {
```
Comment (abhinavsingh1196) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/creator/plan/steps/K8sBGSwapServicesStepPlanCreator.java:70]: nit: redundant else, can just be 
{IF -> RETURN X} RETURN Y;

**Review by abhinavsingh1196 — APPROVED** [review]

File: 125-cd-nextgen/src/main/java/io/harness/cdng/creator/plan/steps/K8sBGSwapServicesStepPlanCreator.java (hunk 1 above)
Comment (alebuh) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/creator/plan/steps/K8sBGSwapServicesStepPlanCreator.java]: done

**Review by GoelPratyush99 — APPROVED** [review]

**Comment by alebuh** [comment]
trigger uts

**Review by abhinavsingh1196 — APPROVED** [review]

**Review by markoharness — APPROVED** [review]

**Review by TarunUba1 — APPROVED** [review]

**Review by rafaelmart1ns — APPROVED** [review]

**Review by yogesh-chauhan — APPROVED** [review]

**Comment by alebuh** [comment]
trigger review

**Pull request merged**